### PR TITLE
Bug fix- separated sessions for downloading and uploading

### PR DESCRIPTION
**Pull request by jayoshih**

### CODE REVIEW

File: ricecooker/classes/nodes.py [hunk 1]
```diff
@@ -204,16 +206,20 @@ def __str__(self):
...
-        if self.domain_ns:
-            return self.domain_ns
-        return self.parent.get_domain_namespace()
+        if not self.domain_ns:
+            self.domain_ns = self.parent.get_domain_namespace()
+        return self.domain_ns
 
     def get_content_id(self):
-        return uuid.uuid5(self.get_domain_namespace(), self.source_id)
+        if not self.content_id:
```
Comment (jayoshih) [inline ricecooker/classes/nodes.py]: Retains the content_id to speed up retrieving it

File: ricecooker/classes/nodes.py [hunk 2]
```diff
@@ -301,38 +307,20 @@ class Video(ContentNode):
...
-            self.transcode_to_lower_resolutions()
-
-        super(Video, self).__init__(source_id, title, files=files, **kwargs)
+        super(Video, self).__init__(source_id, title, **kwargs)
 
     def __str__(self):
         metadata = "{0} {1}".format(len(self.files), "file" if len(self.files) == 1 else "files")
         return "{title} ({kind}): {metadata}".format(title=self.title, kind=self.__class__.__name__, metadata=metadata)
 
-    def derive_thumbnail(self):
```
Comment (jayoshih) [inline ricecooker/classes/nodes.py]: No longer need these methods

File: ricecooker/managers/downloader.py [hunk 3]
```diff
@@ -15,6 +16,10 @@
 from ..exceptions import InvalidFormatException, FileNotFoundException
 from le_utils.constants import file_formats, exercises, format_presets
 
+from requests_file import FileAdapter
```
Comment (jayoshih) [inline ricecooker/managers/downloader.py]: Separate session for downloading

File: ricecooker/managers/downloader.py [hunk 4]
```diff
@@ -271,9 +273,9 @@ def download_file(self, path, title, default_ext=None, preset=None, extracted=Fa
                 return self._file_mapping[filename]
 
         # Catch errors related to reading file path and handle silently
-        except (HTTPError, ConnectionError, InvalidURL, UnicodeDecodeError, UnicodeError, InvalidSchema, IOError):
-            self.failed_files += [(path,title)]
-            return False;
+        except (HTTPError, ConnectionError, InvalidURL, UnicodeDecodeError, UnicodeError, InvalidSchema, IOError) as err:
```
Comment (jayoshih) [inline ricecooker/managers/downloader.py]: Keep track of error thrown

File: ricecooker/__init__.py [hunk 5]
```diff
@@ -2,4 +2,4 @@
 
 __author__ = 'Learning Equality'
 __email__ = 'jordan@learningequality.org'
-__version__ = '0.3.141592'
+__version__ = '0.3.1415926'
```
Comment (aronasorman) [inline ricecooker/__init__.py]: This is getting out of ✋ !

File: ricecooker/classes/nodes.py [hunk 6]
```diff
@@ -204,16 +206,20 @@ def __str__(self):
         return "{title} ({kind}): {metadata}".format(title=self.title, kind=self.__class__.__name__, metadata=metadata)
 
     def get_domain_namespace(self):
-        if self.domain_ns:
-            return self.domain_ns
-        return self.parent.get_domain_namespace()
+        if not self.domain_ns:
+            self.domain_ns = self.parent.get_domain_namespace()
+        return self.domain_ns
```
Comment (aronasorman) [inline ricecooker/classes/nodes.py]: nice ka-ching!

File: ricecooker/classes/nodes.py (hunk 1 above)
Comment (aronasorman) [inline ricecooker/classes/nodes.py]: Nice

**Pull request merged**